control-plane/connect-inject/peering_acceptor_controller.go

```diff
@@ -388,9 +388,11 @@ func (r *PeeringAcceptorController) requestsForPeeringTokens(object client.Objec
 		return []ctrl.Request{}
 	}
 	for _, acceptor := range acceptorList.Items {
-		if acceptor.SecretRef().Backend == "kubernetes" {
-			if acceptor.SecretRef().Name == object.GetName() && acceptor.Namespace == object.GetNamespace() {
-				return []ctrl.Request{{NamespacedName: types.NamespacedName{Namespace: acceptor.Namespace, Name: acceptor.Name}}}
+		if acceptor.SecretRef() != nil {
```

Comment:
Saw a panic in the logs when this function is trying to look at the status secret's Backend, and the status is not yet set, so we only want to do this if the status exists.

Reply:
Can we "and" these conditions? ie
`if acceptor.SecretRef() != nil && acceptor.SecretRef().Backend == "kubernetes" {`

this should work because it will not evaluate the second condition before the first one has succeeded.

Reply:
yup good idea!